.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,235 @@
+              script {
+                def isMajor = sh(script: 'git tag|tail -1|cut -f2-3 -d "."|{ read ver; test $ver == "0.0"; }', returnStatus: true)
+                if (isMajor == 0) {
+                  // We need to create a new branch. First get the name of the branch
+                  def newBranchName = sh(script: 'git tag|tail -1|sed s/v//|cut -f1 -d "."|awk \'{print $1".x"}\'', returnStdout: true)
+                  echo "This appears to be a major version. We are creating a new branch for $newBranchName in the apm-agent-java repo on GitHub."
+                  // Now we need to branch
+                  sh(script: "git checkout -b $newBranchName")
+                  // And push
+                  githubEnv()
+                  gitPush() 
+                  // 6.1 Add the new branch to the conf.yaml in the docs repo
+                  input message: "This was a major version release. Please update the conf.yml in the docs repo before continuing", ok "Continue"
+                } else {  // This was a minor release
+                  // 7. If this was a minor release, reset the current major branch (1.x, 2.x etc) to point to the current tag, e.g. git branch -f 1.x v1.1.0
+                  // Determine current tag
+                  def curTag = sh(script: "git tag|tail -1", returnStdout: true)
+                  def targetBranch = sh(script: 'git tag|tail -1|sed s/v//|cut -f1 -d '.'|awk \'{print $1".x"}\'', returnStdout: true)
+                  sh(script: "git branch -f $targetBranch $curTag")
+                  gitPush()
+                }
+              }
+              // Checkout the tag again!
+              sh("git checkout ${BUILD_TAG}")
+            }
+          }
+        }
+        // 8. Update CHANGELOG.asciidoc to reflect version release. Go over PRs or git log and add bug fixes and features. 
+        stage('Generate a proposed CHANGELOG') {
+          steps {
+            githubEnv()
+            generateChangelog(repo: 'apm-agent-java')
+          }
+        }
+        // 9. Go to elastic/apm-agent-java/releases and draft a new release. Provide a link to release notes in documentation as release description.
+        stage('Create GitHub release draft') {
+          steps {
+            dir("${BASE_DIR}"){
+              script {
+                def curTag = sh(script: "git tag|tail -1", returnStdout: true)
+                // Construct the URL with anchor for the release notes
+                // Ex: https://www.elastic.co/guide/en/apm/agent/java/current/release-notes-1.x.html#release-notes-1.13.0
+                def baseUrl = "https://www.elastic.co/guide/en/apm/agent/java/current/release-notes-"
+                def dotX = sh(script: 'git tag|tail -1|sed s/v//|cut -f1 -d "."|awk \'{print $1".x"}\'', returnStdout: true)
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def finalUrl = sh(script: "echo $baseUrl$dotX\.html#release-notes-$bareTag")
```

Comment:
Shell script that returns the URL?

Reply:
Good suggestion! I have [migrated this logic to a stand-alone shell script](b2488935d718ffc84965de047bdd474c2879db24).